doc/arch/adr-002-geoprocessing-caching.md

```diff
@@ -0,0 +1,103 @@
+# 002 - Geoprocessing Caching
+
+## Context
+
+Geoprocessing calls for large shapes can take a long time to complete. While users can draw custom shapes, they can also pick from a list of predefined shapes in the system. Geoprocessing the same shape over and over when multiple users select it is wasteful and unnecessary. By caching the geoprocessing results of predefined shapes, or Well Known Areas of Interest (WKAoIs), we can improve user experience and application performance.
```

Comment:
It sounds like we're confident of where the bottleneck is, so I wouldn't worry about it.